aioshelly/coap.py

```diff
@@ -69,6 +86,74 @@ def socket_init():
     return sock
 
 
+def verify_l2socket_creation_permission():
+    """Create a socket using the scapy configured l2socket.
+
+    Try to create the socket
+    to see if we have permissions
+    since sniff will do it another
+    thread so we will not be able to capture
+    any permission or bind errors.
+    """
+    conf.L2socket()
+
+
+class MulticastQuerier:
+    """Multicast querier management."""
+
+    def __init__(self):
+        """Initialize multicast querier thread."""
+        try:
+            verify_l2socket_creation_permission()
+        except (Scapy_Exception, OSError) as ex:
+            if os.geteuid() == 0:
```

Comment:
Looks like you need special permissions ;-)

<img width="410" alt="image" src="https://user-images.githubusercontent.com/1444314/106607133-40e74c80-6563-11eb-9ed3-def84696e52a.png">

Reply:
Yes, on **`venv`**. I was not thinking about manual installation of HA when I replied ;-)